src/wallet/rpcwallet.cpp

```diff
@@ -3017,7 +3017,34 @@ UniValue listunspent(const JSONRPCRequest& request)
                 const CScriptID& hash = boost::get<CScriptID>(address);
                 CScript redeemScript;
                 if (pwallet->GetCScript(hash, redeemScript)) {
-                    entry.pushKV("redeemScript", HexStr(redeemScript.begin(), redeemScript.end()));
+                    CTxDestination witness_destination;
+                    if (redeemScript.IsPayToWitnessScriptHash() && ExtractDestination(redeemScript, witness_destination)) {
+                        const WitnessV0ScriptHash& whash = boost::get<WitnessV0ScriptHash>(witness_destination);
+                        CScriptID id;
+                        CRIPEMD160().Write(whash.begin(), whash.size()).Finalize(id.begin());
+                        CScript witnessScript;
+                        // For compatibility with signrawtransaction, if we have both a witnessScript
+                        // and redeemScript, return them in an array
+                        UniValue scripts(UniValue::VARR);
+                        scripts.push_back(HexStr(redeemScript.begin(), redeemScript.end()));
+                        if (pwallet->GetCScript(id, witnessScript)) {
+                            scripts.push_back(HexStr(witnessScript.begin(), witnessScript.end()));
+                        }
+                        entry.push_back(Pair("redeemScript", scripts));
```

Comment:
IIRC, Pair push_backs were just cleaned up... :/

Reply:
Yes, this needs a large rebase and cleanup anyway now signrawtransaction was split up, things like this will be done at the same time when I get onto this